### DISCUSSION TITLE
Exploring the possibility of dropping the save/submit workflow in books?

### DISCUSSION THREAD

**Discussion by nktech1135** [frappe/books#428] [General]
I'm wondering if there is any interest in the community or support from the developer to move towards removing the save/submit workflow in docs like sales invoices and purchase invoices?
It seems to me that it would be cleaner and more intuitive for users to not have to click through both save and submit in books. I'd be interested in making this change with the help of the community if it would have support from the developers.
I'd propose that when something like a purchase invoice is created you would just hit save and the invoice would go to a saved status and fields would go to read only states. If the user wants to edit the invoice there is either an edit button or an amend button that would re open the invoice for editing.

Thoughts anyone?

**Comment by 18alantom** [comment]
I like this idea, a few thoughts on this.

There are two main reasons for the _Save → Submit Workflow_
1. **Saving**: An id is assigned to the transaction
2. **Submitting**: Ledger entries are created

**To remove Submit**: An id would have to be assigned to the transaction when a new entry is being created, i.e. in the Draft state. This means the Draft state will be the Saved (editable) state.

I'm into the idea of removing the Submit button, but because of this change, discarded Drafts will cause discarded ids. (Note: ids are not reused to prevent other issues)

This is something that would otherwise happen only on Cancelling a transaction.

p.s. I'm converting this issue into a discussion

**Reply by nktech1135** [reply]
> discarded Drafts will cause discarded ids. 

I'd see 3 possible approaches to this.
1, we don't care about discarded ids and just let them be. This feels a bit cluttered to me, i'd rather take care of the unused ones somehow.
2, We run a cleanup routine based on some sort of trigger to remove all ids with no status.  We could either run this on a scheduler, or run it every time you hit save on an invoice. There are pros and cons to each approach. We could also run it each time the file is saved/closed.
3, there is a separate table in the db for new invoices. hitting save then moves the database record to an invoices table from a new_invoice table.

I think personally i'd prefer some variant of option 2.

Thoughts?
